client.go

```diff
@@ -30,7 +30,7 @@ func NewDialer(client *http.Client, serverURL string) func(string, string) (net.
 			serverURL:    serverURL,
 			readDeadline: intFromTime(time.Now().Add(10 * 365 * 24 * time.Hour)),
 			received:     make(chan *result, 10),
-			closed:       make(chan bool, 1),
+			closed:       make(chan struct{}),
```

Comment:
It's unnecessary to be buffered as a close signal

Reply:
Ha, looks like I was never actually using the closed channel!